### PR TITLE
Remove redundant manual testing pages and add gestures testing page

### PR DESCRIPTION
**Pull request by avpeery**
This PR updates the manual testing pages by removing redundant pages, updating terrain sources, and adding a page dedicated to gestures. This overhaul reduces the amount of manual testing pages from 38 -> 22. I propose we maintain the manual testing pages to be ~20 pages as we move forward. The testing pages may be adapted in the future as new features are added that need to be manually tested.

Based on the SDD for release improvements, all the following features are addressed in these pages: terrain, fog, projection, feature-state, satellite, canvas size, video, markers, pop-ups, heatmap layer, animate a camera, geojson performance (e.g. `setData`), rtl text, fly-to, image source, custom layer, cooperative gestures, drawing a polygon, fill extrusion, geocode, expressions, and all controls. 

The 21 tests are: 

1. **Animate point**: geoJSON `setData` performance (includes requestAnimationFrame)
2. **Filter features within map view**: queryRenderedFeatures performance (includes popup and expressions)
3. **Add a custom style layer**: uses a custom WebGL style layer
4. **Add a geocoder**: support for geocode plugin
5. **Display driving directions**: support for directions plugin
6. **Show drawn polygon area**: support for draw plugin
7. **Swipe between maps**: support for compare plugin
8. **Add support for right-to-left scripts**: support for RTL plugin
9. **Add a heatmap layer**: support heatmap layer (includes expressions)
10. **Add a 3d model on terrain with ThreeJS:** custom layer to add 3D model
11. **Animate a camera along a path**: following along a path is a common use-case (includes animating camera)
12. **Image source**: image source support
13. **Extrusion query**: 3d extrusion query support
14. **Locate the user**: geolocation support
15. **Query terrain elevation**: focus on terrain performance
16. **Add fog**: focus on fog performance with sky layer
17. **Projections**: focus on projections performance (includes toggling styles and satellite)
18. **Feature State**: updating feature state with different projections
19. **Markers**: marker performance with different projections (includes marker with popup attached and navigation control)
20. **Video**: video on map with different projections (includes satellite)
21. **Gestures**: checklist to test gestures (includes canvas size, fullscreen control, and check for browser support)
22. **Preload tiles**: preload tile performance (includes fly-to and ease-to)

## Launch Checklist

 - [x] briefly describe the changes in this PR
 - [x] write tests for all new functionality
 - [x] manually test the debug page
 - [x] apply changelog label ('bug', 'feature', 'docs', etc) or use the label 'skip changelog'

### CODE REVIEW

File: test/release/index.js [hunk 1]
```diff
@@ -82,18 +42,6 @@ const pages = [
...
-        "title": "Add Terrain"
-    },
-    {
-        "key": "atmospheric-sky",
-        "title": "Atmospheric Sky"
-    },
-    {
-        "key": "free-camera-point",
-        "title": "Free Camera Point"
-    },
```
Comment (karimnaaji) [inline test/release/index.js:96]: Do you know where the free camera API is now covered?

File: test/release/index.js [hunk 2]
```diff
@@ -111,36 +59,16 @@ const pages = [
         "key": "locate-user",
         "title": "Locate the user"
     },
-    {
-        "key": "extrusion-query",
-        "url": "./extrusion-query.html",
-        "title": "Fill extrusion querying with terrain"
-    },
```
Comment (karimnaaji) [inline test/release/index.js:118]: I'm not sure if fill-extrusion querying with terrain is tested elsewhere and truly redundant, but I also wonder if the testing coverage of query tests to be sufficient for this one. It's probably ok to remove.

File: test/release/index.js [hunk 3]
```diff
@@ -111,36 +59,16 @@ const pages = [
...
     {
         "key": "3d-playground",
         "title": "3D Playground",
         "url": "./3d-playground.html"
     },
```
Comment (karimnaaji) [inline test/release/index.js:66]: I'd be in favor of splitting that one up in several more focused examples:
- Sky: toggle between gradient and atmosphere 
- Terrain: A default terrain page with a style switcher
- Free camera: A page utilizing the free camera API

The issue from this page is that the UI isn't really appropriate for toggling things on mobile. It was useful for v2 but in retrospect it's not clear what should be tested for this page as it's a bit too generic.

**Review by karimnaaji — COMMENTED** [review]
Thanks for tackling that @avpeery . I've got a few suggestions and questions to make sure we're not missing out on important features to test.

File: test/release/index.js [hunk 4]
```diff
@@ -6,58 +6,18 @@
...
-        "title": "Add GeoJSON marker"
-    },
     {
         "key": "animate-point-along-line",
         "title": "Animate point"
     },
-    {
-        "key": "queryrenderedfeatures",
-        "title": "Get features under the mouse pointer"
-    },
```
[the remark this answers is not in the document]
Comment (avpeery) [reply test/release/index.js:20]: I've included the filter within map view example, which includes feature querying and popups: https://docs.mapbox.com/mapbox-gl-js/example/filter-features-within-map-view/.

File: test/release/index.js (hunk 1 above)
Comment (avpeery) [reply test/release/index.js:96]: The free camera api is included in animate a camera along path: https://docs.mapbox.com/mapbox-gl-js/example/free-camera-path (and 3d playground)

File: test/release/index.js (hunk 2 above)
Comment (avpeery) [reply test/release/index.js:118]: fill-extrusion querying with terrain is included in 3d-playground and fog

File: test/release/index.js (hunk 1 above)
Comment (karimnaaji) [reply test/release/index.js:96]: Thanks for confirming!

File: test/release/index.js (hunk 2 above)
Comment (karimnaaji) [reply test/release/index.js:118]: >fill-extrusion querying with terrain is included in 3d-playground and fog

I only see terrain elevation querying in those pages?

Comment (avpeery) [reply test/release/index.js:118]: This is my mistake. just setting fill extrusion, not querying in those pages. We can add this back in and split up 3d-playground in favor of more straight forward testing pages for terrain and sky.

Comment (karimnaaji) [reply test/release/index.js:118]: No worries, I checked and it's covered by some query test so a good call to remove.

**Comment by avpeery** [comment]
@karimnaaji Removed 3d-playground and fog pages. These pages are not friendly for testing on mobile devices and make the testing pages more complex. Instead, examples add fog (includes fog and sky layer) and query terrain elevation (queries terrain elevation with popup) were added. Since removing the other pages removed covering extrusion, extrusion-query was re-added.

**Pull request merged**